x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -486,6 +487,7 @@ func s3ObjectHash(obj s3EventV2) string {
 	h := sha256.New()
 	h.Write([]byte(obj.S3.Bucket.ARN))
 	h.Write([]byte(obj.S3.Object.Key))
+	h.Write([]byte(obj.S3.Object.LastModified.String()))
```

Comment:
I suggest writing the unix nano value into the hash instead of the formatted time string. It's immune to any time formatting or time zone issues, and it avoids any allocations.

```
	var ts [8]byte
	binary.LittleEndian.PutUint64(ts[:], uint64(obj.S3.Object.LastModified.UnixNano()))
	h.Write(ts[:])
```

```
BenchmarkTime/unix-nano-to-array-10         	267578316	         4.381 ns/op	       0 B/op	       0 allocs/op
BenchmarkTime/time-string-10                	 4762586	       251.1 ns/op	     192 B/op	       4 allocs/op
```

Reply:
But upon futher consideration, perhaps it is even better[^1] to simply prefix the `_id` with the time

`{{last-modified unix nanos}}-{{existing-bucket+key-hash}}-{{offset}}`

[^1]: https://www.elastic.co/blog/efficient-duplicate-prevention-for-event-based-data-in-elasticsearch